org.eclipse.jdt.core.manipulation/refactoring/org/eclipse/jdt/internal/corext/refactoring/structure/PushDownRefactoringProcessor.java

```diff
@@ -375,14 +375,80 @@ private static IJavaElement[] getReferencingElementsFromSameClass(IMember member
 		return result.toArray(new IJavaElement[result.size()]);
 	}
 
+	private static IJavaElement[] getReferencingElementsFromProject(IMember member, IProgressMonitor pm, RefactoringStatus status) throws JavaModelException {
+		Assert.isNotNull(member);
+		SearchPattern pattern= SearchPattern.createPattern(member,
+				IJavaSearchConstants.REFERENCES, SearchUtils.GENERICS_AGNOSTIC_MATCH_RULE);
+		if (pattern == null) {
+			return new IJavaElement[0];
+		}
+		final RefactoringSearchEngine2 engine= new RefactoringSearchEngine2(pattern);
+		engine.setFiltering(true, true);
+		IType declaringType= member.getDeclaringType();
+		engine.setScope(SearchEngine.createJavaSearchScope(new IJavaElement[] { member.getJavaProject() }));
+		engine.setStatus(status);
+		engine.searchPattern(pm);
+		Set<IJavaElement> result= new HashSet<>(3);
+		ICompilationUnit cu= member.getCompilationUnit();
+		String source= cu.getSource();
+		CompilationUnit cuRoot= null;
+		for (SearchResultGroup group : (SearchResultGroup[]) engine.getResults()) {
+			for (SearchMatch searchResult : group.getSearchResults()) {
+				if (source.charAt(searchResult.getOffset() - 1) == '.') {
```

Comment:
@jjohnstn  this line looks wrong to me: there is no guarantee that `searchResult` denotes a location within `source` (in fact I'm seeing `StringIndexOutOfBoundsException` right here).

I believe instead of fetching `source` from `member`  the outer loop should start with
```
		source = group.getCompilationUnit().getSource();
```

Reply:
@stephan-herrmann I'll look at it  Do you have a test scenario?

Reply:
Looked at it.  You are right.  The source should have been changed per SearchResultGroup.  Thanks.

Reply:
Created https://github.com/eclipse-jdt/eclipse.jdt.ui/pull/2151 to fix

Reply:
Patch merged.

Reply:
@stephan-herrmann Marking as resolved.